### PR TITLE
[HOPSWORKS-1425] fix hopsworks to work with hopsfs scheme"

### PR DESCRIPTION
**Pull request by berthoug**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1425

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the new behavior (if this is a feature change)?**
hopsfs changed to use hopsfs as default scheme. Hopsworks need to be able to handle this scheme and not expect only hdfs for scheme.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
Depends on hops-2.8.2.9-SNAPSHOT.
For this to work with chef the following pull request need to be merged:
logicalclocks/hive-chef#37
logicalclocks/spark-chef#84
logicalclocks/hops-hadoop-chef#171
logicalclocks/hops-util-py#138
logicalclocks/hive#5

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/util/PathValidator.java [hunk 1]
```diff
@@ -94,7 +94,7 @@
   public DsPath validatePath(Project project, String path) throws DatasetException, ProjectException {
     DsPath dsPath = new DsPath();
 
-    Pattern authorityPattern = Pattern.compile("(hdfs://[a-zA-Z0-9\\-\\.]{2,255}:[0-9]{4,6})(/.*$)");
+    Pattern authorityPattern = Pattern.compile("([a-zA-Z0-9]+://[a-zA-Z0-9\\-\\.]{2,255}:[0-9]{4,6})(/.*$)");
```
Comment (ErmiasG) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/util/PathValidator.java]: Use Utils.prepPath(path)

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/serving/ServingUtil.java [hunk 2]
```diff
@@ -49,15 +50,17 @@
   private InodeController inodeController;
   
   /**
-   * Validates user input before creating or updating a serving. This method contains the common input validation
-   * between different serving types and then delegates to serving-type-specific controllers to do validation specific
-   * to the serving type.
+   * Validates user input before creating or updating a serving.This method contains the common input validation
+ between different serving types and then delegates to serving-type-specific controllers to do validation specific
```
Comment (ErmiasG) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/serving/ServingUtil.java]: indentation

**Review by ErmiasG — CHANGES_REQUESTED** [review]

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/util/PathValidator.java (hunk 1 above)
Comment (berthoug) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/util/PathValidator.java]: fixed

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/serving/ServingUtil.java (hunk 2 above)
Comment (berthoug) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/serving/ServingUtil.java]: fixed

**Pull request merged**